src/modules/twap/updaters/SettledPartOrdersUpdater.tsx

```diff
@@ -0,0 +1,76 @@
+import { useAtomValue } from 'jotai'
+import { useUpdateAtom } from 'jotai/utils'
+import { useEffect, useMemo } from 'react'
+
+import { Order } from 'legacy/state/orders/actions'
+import { useAddOrUpdateOrders } from 'legacy/state/orders/hooks'
+
+import { useSWRProdOrders } from 'modules/orders/hooks/useSWRProdOrders'
+import { tokensByAddressAtom } from 'modules/tokensList/state/tokensListAtom'
+import { useWalletInfo } from 'modules/wallet'
+
+import { twapOrdersAtom } from '../state/twapOrdersListAtom'
+import { markPartOrdersAsSettledAtom, TwapPartOrderItem, twapPartOrdersListAtom } from '../state/twapPartOrdersAtom'
+import { mapPartOrderToStoreOrder } from '../utils/mapPartOrderToStoreOrder'
+
+const isVirtualPart = false
+
+/**
+ * For complete control of discrete orders created by TWAP, we process them separately from other discrete orders
+ * Since WatchTower creates orders only in PROD env, we use useSWRProdOrders()
+ * To distinguish parts settled in order-book from other parts, we mark them by isSettledInOrderBook flag
+ */
+export function SettledPartOrdersUpdater() {
+  const { chainId } = useWalletInfo()
+  const prodOrders = useSWRProdOrders()
+  const tokensByAddress = useAtomValue(tokensByAddressAtom)
+  const twapPartOrdersList = useAtomValue(twapPartOrdersListAtom)
+  const twapOrders = useAtomValue(twapOrdersAtom)
+  const markPartOrdersAsSettled = useUpdateAtom(markPartOrdersAsSettledAtom)
+  const addOrUpdateOrders = useAddOrUpdateOrders()
+
+  const twapPartOrdersMap = useMemo(() => {
+    return twapPartOrdersList.reduce<{ [id: string]: TwapPartOrderItem }>((acc, val) => {
+      acc[val.uid] = val
+      return acc
+    }, {})
+  }, [twapPartOrdersList])
+
+  // Take only orders related to TWAP from prod API response
+  const partOrdersFromProd = useMemo(() => {
+    return prodOrders
+      .filter((enrichedOrder) => {
+        const item = twapPartOrdersMap[enrichedOrder.uid]
+        const parent = twapOrders[item?.twapOrderId]
+
+        return !!parent
+      })
+      .map<Order>((enrichedOrder) => {
+        const item = twapPartOrdersMap[enrichedOrder.uid]
+        const parent = twapOrders[item.twapOrderId]
+
+        return mapPartOrderToStoreOrder(item, enrichedOrder, isVirtualPart, parent, tokensByAddress)
+      })
+  }, [prodOrders, twapPartOrdersMap, tokensByAddress, twapOrders])
+
+  useEffect(() => {
+    if (!partOrdersFromProd.length) return
+
+    const settledOrders = partOrdersFromProd.reduce<{ [parentId: string]: string[] }>((acc, val) => {
+      const parentId = val.composableCowInfo?.parentId
+
+      if (parentId) {
+        acc[parentId] = acc[parentId] || []
+
+        acc[parentId].push(val.id)
+      }
+
+      return acc
+    }, {})
+
+    markPartOrdersAsSettled(settledOrders)
```

Comment:
I don't understand why they are marked as settled.
AFAICT, the part orders have been returned by the API, but there's no check for the order status.
Or there is and I missed it?

Reply:
In this context `settled` means have been created in the order-book

Reply:
The flag is named `isSettledInOrderBook`

Reply:
So by settled you mean created?

Reply:
Yes :)

Reply:
Unless this is part of the contract, is it possible to rename it then?
Settled to me means fulfilled and I imagine I won't be the only one confused.